core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -311,6 +311,7 @@ class SparkContext(config: SparkConf) extends Logging {
    *  in case of local spark app something like 'local-1433865536131'
    *  in case of YARN something like 'application_1433865536131_34483'
    *  in case of MESOS something like 'driver-20170926223339-0001'
+   *  in case of Kubernetes something like 'spark-application-1559610575898'
```

Comment:
@skonto is this correct?

Reply:
@srowen This is not true for cluster mode. In cluster mode application_id is set [here](https://github.com/apache/spark/blob/b30655bdef71f1aa85811464a9f6b61057549664/core/src/main/scala/org/apache/spark/SparkContext.scala#L554) in the Spark Context and this value is retrieved from [here](https://github.com/apache/spark/blob/b30655bdef71f1aa85811464a9f6b61057549664/core/src/main/scala/org/apache/spark/scheduler/TaskScheduler.scala#L97). But it does not default to:
```
  private val appId = "spark-application-" + System.currentTimeMillis
```
as it is overridden in [here](https://github.com/apache/spark/blob/b30655bdef71f1aa85811464a9f6b61057549664/core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala#L900):
```
  override def applicationId(): String = backend.applicationId()
```
The K8s backend will set that value [here](https://github.com/apache/spark/blob/b30655bdef71f1aa85811464a9f6b61057549664/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala#L74):
```
 conf.getOption("spark.app.id").map(_.toString).getOrElse(super.applicationId)
``` 
For cluster mode the `spark.app.id` is always set [here](https://github.com/apache/spark/blob/b30655bdef71f1aa85811464a9f6b61057549664/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala#L207) to be:
```
val kubernetesAppId = s"spark-${UUID.randomUUID().toString.replaceAll("-", "")}"
```
An example:   Spark Application Id:                       spark-3d4ef94a42e1428c9658f13573abfa38